upstairs/src/volume.rs

```diff
@@ -4310,4 +4437,604 @@ mod test {
+                        extent_count: 32,
+                        gen: 2,
+                        opts: CrucibleOpts {
+                            id: Uuid::new_v4(),
+                            target: vec![
+                                "[fd00:1122:3344:103::7]:19002"
+                                    .parse()
+                                    .unwrap(),
+                                "[fd00:1122:3344:101::7]:19002"
+                                    .parse()
+                                    .unwrap(),
+                                "[fd00:1122:3344:102::8]:19002"
+                                    .parse()
+                                    .unwrap(),
+                            ],
+                            ..Default::default()
+                        },
+                    }],
+                    read_only_parent: Some(Box::new(
+                        VolumeConstructionRequest::Volume {
+                            block_size: 512,
+                            id: Uuid::new_v4(),
+                            sub_volumes: vec![
+                                VolumeConstructionRequest::Region {
+                                    block_size: 512,
+                                    blocks_per_extent: 131072,
+                                    extent_count: 32,
+                                    gen: 1,
+                                    opts: CrucibleOpts {
+                                        id: Uuid::new_v4(),
+                                        target: vec![
+                                            "[fd00:1122:3344:203::7]:19002"
+                                                .parse()
+                                                .unwrap(),
+                                            "[fd00:1122:3344:201::7]:19002"
+                                                .parse()
+                                                .unwrap(),
+                                            "[fd00:1122:3344:202::8]:19002"
+                                                .parse()
+                                                .unwrap(),
+                                        ],
+                                        ..Default::default()
+                                    },
+                                },
+                            ],
+                            read_only_parent: None,
+                        },
+                    )),
+                },
+            )),
+        };
+
+        // Replace one of the deeper ROP's subvolume targets, and don't bump the
+        // gen
+        let mut replacement = original.clone();
+        match &mut replacement {
+            VolumeConstructionRequest::Volume {
+                sub_volumes,
+                read_only_parent,
+                ..
+            } => {
+                match &mut sub_volumes[0] {
+                    VolumeConstructionRequest::Region { gen, .. } => {
+                        *gen += 1;
```

Comment:
If we don't bump any gen, will this still pass?

Reply:
yep - added volume_replace_rop_changes_same_gen in e0fafe78